pkg/kubelet/cm/cpuset/cpuset.go

```diff
@@ -301,7 +301,7 @@ func Parse(s string) (CPUSet, error) {
 	ranges := strings.Split(s, ",")
 
 	for _, r := range ranges {
-		boundaries := strings.Split(r, "-")
+		boundaries := strings.SplitN(r, "-", 2)
```

Comment:
Does the cpu value need to be non-negative? The range `-2--1` will parse correctly as `-2,-1`.

Reply:
Good point!  That's another hole I hadn't thought of - Every int we parse must be >=0

Reply:
Actually, turns out we get this for free!

Because of `SplitN(r, "-", 2)`, trying to parse `"-2--1"` will split exactly once into `""` and `"2--1"`, neither of which will get past the subsequent `strconv.Atoi()`.  In fact, this catches anything that starts with a `-`, making other potential odd cases like `"--1"` likewise invalid.

That said, I think it's valuable adding both `"-2--1"` and `"--1"` to the negative test list to ensure this doesn't break in the future.

Reply:
Added.